core/src/main/java/org/apache/carbondata/core/scan/collector/impl/RestructureBasedVectorResultCollector.java

```diff
@@ -238,7 +238,7 @@ private void fillDataForNonExistingMeasures() {
                 (long) defaultValue);
           } else if (DataTypes.isDecimal(dataType)) {
             vector.putDecimals(columnVectorInfo.vectorOffset, columnVectorInfo.size,
-                (Decimal) defaultValue, measure.getPrecision());
+                ((Decimal) defaultValue).toJavaBigDecimal(), measure.getPrecision());
```

Comment:
please remove these import : 
 import org.apache.spark.sql.types.Decimal;
 import org.apache.spark.unsafe.types.UTF8String;

Reply:
We can not remove these imports as the data that is  returned is stored in Spark Format , the casting is done to convert it to generic types

Reply:
ok, this is core module.